src/common/Analytics/AnalyticsEvents.ts

```diff
@@ -0,0 +1,91 @@
+import { EventEmitter } from 'events';
+
+export type AnalyticsEventTypeHandlers = {
+  track: (eventName: string, event: unknown) => void;
+  identifyUser: (id: string) => void;
+  userPropertyUpdate: (
+    key: string,
+    value: string | number | boolean | undefined,
+  ) => void;
+  reset: () => void;
+};
+
+export type AnalyticsEventTypes = keyof AnalyticsEventTypeHandlers;
+export type AnalyticsEventTypeHandler<T extends AnalyticsEventTypes> =
+  AnalyticsEventTypeHandlers[T];
+
+class AnalyticsEvents {
+  private emitter = new EventEmitter();
+
+  public addListener<EventType extends AnalyticsEventTypes>(
+    eventType: EventType,
+    listener: AnalyticsEventTypeHandler<EventType>,
+  ) {
+    return this.emitter.addListener(eventType, listener);
+  }
+
+  public removeListener<EventType extends AnalyticsEventTypes>(
+    eventType: EventType,
+    listener: AnalyticsEventTypeHandler<EventType>,
+  ) {
+    return this.emitter.removeListener(eventType, listener);
+  }
+
+  public emit<EventType extends AnalyticsEventTypes>(
+    eventType: EventType,
+    ...params: Parameters<AnalyticsEventTypeHandler<EventType>>
+  ) {
+    return this.emitter.emit(eventType, ...params);
+  }
+}
```

Comment:
I'd recommend removing this `class` altogether, since we're not supporting creating multiple instances. Instead, we can just do this:

```typescript
const emitter = new EventEmitter();

// reference `emitter` in `createAnalyticsEmitter` etc.

type AnalyticsEmitter = {
  addListener<EventType extends AnalyticsEventTypes>(
    eventType: EventType,
    listener: AnalyticsEventTypeHandler<EventType>,
  ): void;

  removeListener<EventType extends AnalyticsEventTypes>(
    eventType: EventType,
    listener: AnalyticsEventTypeHandler<EventType>,
  ): void;
};

export const analyticsListener = {
  addListener: (...args) => emitter.addListener(...args),
  removeListener: (...args) => emitter.addListener(...args)
}
```

I think that will slim down the LOC here quite a lot